src/datafusion.rs

```diff
@@ -26,21 +25,29 @@ use std::path::Path;
 use std::sync::{Arc, Mutex};
 
 pub struct DbaseTable {
+    path: String,
     reader: Arc<Mutex<Reader<BufReader<std::fs::File>>>>,
 }
 
 impl Clone for DbaseTable {
     fn clone(&self) -> Self {
         return DbaseTable {
+            path: self.path.clone(),
             reader: self.reader.clone(),
         };
     }
 }
 
 impl DbaseTable {
-    pub fn new<P: AsRef<Path>>(path: P) -> Self {
-        let reader = Reader::from_path(path).unwrap();
+    pub fn new<P: AsRef<Path> + Debug>(path: P) -> Self {
+        let reader = Reader::from_path(&path)
+            .expect(format!("Could not find file {:?} or corresponding memo file", &path).as_str());
         return DbaseTable {
+            path: path
+                .as_ref()
+                .to_str()
+                .expect("Failed to convert path to string")
```

Comment:
A slightly more descriptive error could be "Path contains non-unicode characters"

Reply:
done